packages/filesystem/src/browser/file-tree/file-tree-widget.tsx

```diff
@@ -152,14 +162,20 @@ export class FileTreeWidget extends TreeWidget {
             }
         }
     }
-
-    protected setTreeNodeAsData(data: DataTransfer, node: TreeNode): void {
-        data.setData('tree-node', node.id);
+    protected setTreeNodeAsData(data: DataTransfer, sourceNode: TreeNode, relatedNodes: TreeNode[]): void {
```

Comment:
the last argument should be optional, otherwise it is a breaking change for subclasses calling this method without 3rd arg

Reply:
or we add a new method instead, i.e. `setSelectedTreeNodesAsData`, which calls to `setTreeNodeAsData` in addition to setting `selected-tree-nodes`, and leave original method without modifications

Reply:
I think the second implementation is a bit better, but I think the original method should be marked as deprecated. With reference to vscode, their drag event data is always an array, not passing the original element that generated the event. So I don't think the node array is an optional parameter and should be used as the main parameter.

Reply:
marking deprecated and adding depreciation, i.e since when and what to use instead, sounds fine